File: 58.length-of-last-word.js
/*
 * @lc app=leetcode id=58 lang=javascript
 *
 * [58] Length of Last Word
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s)
{
    s = s.trimEnd();
    let right = s.length - 1;
    while (right >= 0
        && s[right] !== ' ')
    {
        right--;
    }

    return s.length - right - 1;
};
// @lc code=end

File: src/2023-2024/58.length-of-last-word.cpp
/*
 * @lc app=leetcode id=58 lang=cpp
 *
 * [58] Length of Last Word
 */
#include <string>

using std::string;

// @lc code=start
class Solution {
 public:
  int lengthOfLastWord(string s) {
    int index = s.length() - 1;
    while (index >= 0 && s[index] == ' ') {
      index--;
    }

    int wordLength = 0;
    while (index >= 0 && s[index] != ' ') {
      wordLength++;
      index--;
    }

    return wordLength;
  }
};
// @lc code=end
